File: backend/shared/src/main/scala/delivery/model/Profiles.scala
package delivery.model

/** 与前端 `frontend/src/delivery/model/profiles.ts` 对齐 */

final case class CustomerProfile(
    id: String,
    name: String,
    phone: String,
    defaultAddress: String,
    vouchers: List[Voucher],
    walletBalance: Double,
    pendingOrders: List[Order],
    historyOrders: List[Order]
)

final case class MerchantStoreProfile(
    merchant: Merchant,
    products: List[Product],
    pendingOrders: List[Order],
    historyOrders: List[Order]
)

final case class MerchantProfile(
    id: String,
    ownerName: String,
    phone: String,
    stores: List[MerchantStoreProfile]
)

final case class RiderProfile(
    rider: Rider,
    walletBalance: Double,
    pendingOrders: List[Order],
    historyOrders: List[Order]
)

File: frontend/src/delivery/model/profiles.ts
/** 与后端 `Profiles.scala` 对齐 */

import type { Merchant, Product } from './catalog'
import type { Order } from './order'
import type { Rider } from './rider'
import type { Voucher } from './voucher'

export interface CustomerProfile {
  id: string
  name: string
  phone: string
  defaultAddress: string
  vouchers: Voucher[]
  walletBalance: number
  pendingOrders: Order[]
  historyOrders: Order[]
}

export interface MerchantStoreProfile {
  merchant: Merchant
  products: Product[]
  pendingOrders: Order[]
  historyOrders: Order[]
}

export interface MerchantProfile {
  id: string
  ownerName: string
  phone: string
  stores: MerchantStoreProfile[]
}

export interface RiderProfile {
  rider: Rider
  walletBalance: number
  pendingOrders: Order[]
  historyOrders: Order[]
}
